http/1.7/modules/ROOT/pages/http-polling-source-ref.adoc

```diff
@@ -0,0 +1,300 @@
+        </when>
+        <when expression="#[attributes.queryParams.watermark == '5']">
+            <set-payload value="#[output application/json --- {'items': [{'name': 'Ivan', 'wm': 6}, {'name': 'Hyeran', 'wm': 7}]}]"/>
+        </when>
+        <otherwise>
+            <set-payload value="#[output application/json --- {'items': []}]"/>
+        </otherwise>
+    </choice>
+</flow>
+
+<flow name="watermarkIntoQueryParamsPollingFlow">
+    <http:polling-source config-ref="watermarkRequestConfig" path="watermark-into-query"
+        splitExpression="#[payload.items]" watermarkExpression="#[item.wm]">
+        <scheduling-strategy>
+            <fixed-frequency frequency="1" timeUnit="SECONDS"/>
+        </scheduling-strategy>
+        <http:polling-request-query-params >
+            <http:polling-request-query-param key="watermark" value="#[watermark default 0]" />
+        </http:polling-request-query-params>
+    </http:polling-source>
+    <logger message="#[payload]"/>
+</flow>
+----
+
+== Configure Idempotency Expression
+
+To avoid the concurrent execution of two flows with the same payload identification, configure idempotency for the HTTP *Polling Source* event source. The configuration ensures that payloads with the same ID are processed not concurrently.
+
+To configure idempotency for the event source, add an ID expression to retrieve the ID from the item, similar to the watermarking expression configuration. Use `payload` and `item` placeholders for the idempotency expression `idExpression`.
+
+The following XML configuration example shows that the first three items execute the flow concurrently, and the fourth item starts only its process after the first item, both with the same ID, finishes processing.
+
+[source,xml,linenums]
+----
+<flow name="identityWithoutWatermarkListenerFlow">
+    <http:listener config-ref="watermarkListenerConfig" path="/identity-no-watermark"/>
+    <set-payload value="#[output application/json --- {'items': [{'name': 'Rodro', 'value': 5}, {'name': 'Eze', 'value': 8}, {'name': 'MG', 'value': 7}, {'name': 'Rodro', 'value': 14}]}]"/>
+</flow>
+
+<flow name="identityWithoutWatermarkPollingFlow">
+    <http:polling-source config-ref="watermarkRequestConfig" path="identity-no-watermark"
+        splitExpression="#[payload.items]" idExpression="#[item.name]">
+        <scheduling-strategy>
+            <fixed-frequency frequency="1" timeUnit="HOURS"/>
+        </scheduling-strategy>
+    </http:polling-source>
+    <logger message="#[payload]"/>
+</flow>
+----
+
+
+
+== Configure Response Validator
+
+To consider whether the response is valid, configure a response validator, as you do with the HTTP *Request* operation.
+
+For more information, refer to the xref:http-request-ref#HTTP-Response-Validation[Configure Response Validator] in the HTTP *Request* operation documentation.
```

Comment:
I would omit this: in previous comparisons between event source and HTTP request operation, we have not included cross-refs.

As I said earlier, repeating these comparisons between the two might be adding unnecessary extra "noise" to the text focusing on the event source. Consider stating one time, in the H1 intro to these H2 topics, that you configure an event source in much the same was as an HTTP request, and then _maybe_ include a cross-ref to HTTP request configuration as a whole.

Reply:
```suggestion
```